hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/AvroConvertor.java

```diff
@@ -108,9 +109,13 @@ private void initJsonConvertor() {
   }
 
   public GenericRecord fromJson(String json) {
-    initSchema();
-    initJsonConvertor();
-    return jsonConverter.convert(json, schema);
+    try {
+      initSchema();
+      initJsonConvertor();
+      return jsonConverter.convert(json, schema);
+    } catch (Exception e) {
+      throw new HoodieDeltaStreamerSchemaCompatibilityException("Failed to convert schema from json to avro", e);
```

Comment:
We want to distinguish between parsing/conversion style errors and compatibility errors. Compatibility is more like something violates the supported schema evolution [here](https://hudi.apache.org/docs/schema_evolution#out-of-the-box-schema-evolution) 

For all of the errors in this class, they fall into a parsing/conversion style error